slides/03-deploy.qmd

```diff
@@ -141,26 +141,464 @@ vetiver.predict(endpoint=endpoint, data=X_test)
 - [Ten simple rules for writing Dockerfiles for reproducible data science](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1008316)
 - [Docker info from Posit Solutions Engineering](https://solutions.posit.co/envs-pkgs/environments/docker/)
 
-# Deploy preprocessors and models together
-
-tktk more substantive feature engineering (like with learned transformation from training data)
+# Deploy preprocessors and models together 🤗
 
 ## Deploy preprocessors and models together
 
 ![](images/good_workflow.png){fig-align="center"}
 
-
 ## What is wrong with this?
 
 ![](images/bad_workflow.png){fig-align="center"}
 
-# Add a new endpoint to your API
+## Deploy preprocessors and models together
+
+::: panel-tabset
+### R
+
+```{r}
+library(tidyverse)
+library(tidymodels)
+library(arrow)
+path <- here::here("data", "inspections.parquet")
+inspections <- read_parquet(path)
+
+set.seed(123)
+inspect_split <- initial_split(inspections, prop = 0.8)
+inspect_train <- training(inspect_split)
+inspect_test <- testing(inspect_split)
+```
+
+
+### Python
+
+```{python}
+import pandas as pd
+import numpy as np
+from sklearn import preprocessing, ensemble, pipeline, compose, model_selection
+
+inspections = pd.read_parquet('../data/inspections.parquet')
+inspections['inspection_date'] = pd.to_datetime(inspections['inspection_date'])
+inspections['month'] = inspections['inspection_date'].dt.month
+inspections['year'] = inspections['inspection_date'].dt.year
+
+categorical_features = ['facility_type', 'risk', 'month', 'year']
+X, y = inspections.drop(columns=['aka_name', 'results','inspection_date']), inspections['results']
+X_train, X_test, y_train, y_test = model_selection.train_test_split(
+    X, y,
+    stratify = y,
+    test_size = 0.2
+)
+```
+:::
+
+## Deploy preprocessors and models together
+
+::: panel-tabset
+### R
+
+```{r}
+inspection_rec <- 
+  recipe(results ~ facility_type + risk + total_violations + inspection_date, 
+         data = inspect_train) |> 
+  step_date(inspection_date, features = c("month", "year"), keep_original_cols = FALSE) |> 
+  step_dummy(all_nominal_predictors()) |> 
+  step_normalize(all_predictors())
+
+inspection_fit <-
+  workflow(inspection_rec, svm_rbf(mode = "classification")) |> 
+  fit(data = inspect_train)
+```
+
+### Python
+
+```{r}
+## TODO for Isabel: put a model here that involves substantive feature engineering (i.e. something learned that could cause data leakage)
```

Comment:
Here is what we talked about in terms of a new model that has more meaningful feature engineering. Alternatively, we could just cut this section. This 3rd section is pretty long and I don't know that having this in here is really that useful compared to what else we could do. 

If we cut it, I would probably move those couple of slides about deploying feature engineering + model together to slide deck 1 when we are first making a vetiver model object.

Reply:
I think it makes more sense to talk about deploying feature engineering + model together in slide deck one, especially since those slides talk explicitly about the model lifecycle stages (including feature engineering).

Reply:
Done!